input-profile.js

```diff
@@ -0,0 +1,249 @@
+import {Component, Property} from '@wonderlandengine/api';
+import {vec3, quat} from 'gl-matrix';
+const _TempVec = vec3.create();
+const _TempQuat = quat.create();
+
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+    static Properties = {
+        controller: Property.object(),
+        handednessIndex: Property.enum(['left', 'right'], 0),
+        path: Property.string(
+            'https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@1.0.13/dist/profiles/'
+        ),
+        customProfileFolder: Property.string(''),
+        preSpawnedController: Property.object(),
+    };
+
+    init() {
+        this._gamepadObjects = {};
+        this._modelLoaded = false;
+        this.controllerDeactivateFlag = false;
+        this.ProfileJSON = {};
+    }
+
+    start() {
+        if (this.engine.XRSession) {
+            this.engine.XRSession.addEventListener(
+                'inputsourceschange',
+                this.onInputSourcesChange.bind(this)
+            );
+        } else {
+            this.engine.onXRSessionStart.add((session) => {
+                session.addEventListener(
+                    'inputsourceschange',
+                    this.onInputSourcesChange.bind(this)
+                );
+            });
+        }
+
+        this.handedness = ['left', 'right'][this.handednessIndex];
+
+        this.components = [];
+        if (this.preSpawnedController != null) {
+            this.getComponents(this.preSpawnedController);
+        }
+    }
+
+    update() {
+        if (this.modelLoaded) {
```

Comment:
Because `update()` is so simple, you could basically just assign the function in `loadModel` instead:

```js
loadModel() {
    this.update = (() => this.mapGamepadInput());
}
```